primitives/src/v1/mod.rs

```diff
@@ -1349,9 +1349,41 @@ pub struct DisputeStatementSet {
 	pub statements: Vec<(DisputeStatement, ValidatorIndex, ValidatorSignature)>,
 }
 
+impl AsRef<DisputeStatementSet> for DisputeStatementSet {
+	fn as_ref(&self) -> &DisputeStatementSet {
+		&self
+	}
+}
+
 /// A set of dispute statements.
 pub type MultiDisputeStatementSet = Vec<DisputeStatementSet>;
 
+/// A _checked_ set of dispute statements.
+#[derive(Clone)]
+pub struct CheckedDisputeStatementSet(DisputeStatementSet);
```

Comment:
Should we use an enum with variants: Checked/Unchecked ?

Reply:
No, that would remove type level guarantees that two disjunctive types allow.